ai-eval-service/evaluator.py

```
@@ -361,18 +251,22 @@ async def evaluate_batch(
-            langfair_category = category_mapping.get(item['category'].lower(), "gender")
+            category_lower = item['category'].lower()
+            if category_lower not in self.CATEGORY_MAPPING:
+                frame = inspect.currentframe()
+                context_info = ""
+                if frame:
+                    filename = frame.f_code.co_filename
+                    func_name = frame.f_code.co_name
+                    line_no = frame.f_lineno
+                    context_info = f" (file: {filename}, function: {func_name}, line: {line_no})"
+                logger.warning(
+                    f"Unrecognized category '{item['category']}' not found in CATEGORY_MAPPING, defaulting to 'gender'.{context_info}"
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)